workspaces/config/lib/set-envs.js

```diff
@@ -101,10 +101,7 @@ const setEnvs = (config) => {
   if (cliConf['node-options']) {
     env.NODE_OPTIONS = cliConf['node-options']
   }
-
-  if (require.main && require.main.filename) {
-    env.npm_execpath = require.main.filename
-  }
+  env.npm_execpath = config.npmBin
   env.NODE = env.npm_node_execpath = config.execPath
```

Comment:
We are setting process.env.NODE here, this could be why we later short circuit to reading back from it in the `nodeBin` variable.  This is another place to reevaluate in npm 11 or beyond.